### PR TITLE
unstructured 0.15.8 ❄️

### PR DESCRIPTION
**Pull request by lorepirri**
unstructured 0.15.8 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-4794]
- dev_url (pypi): https://github.com/Unstructured-IO/unstructured/tree/0.15.8
- conda-forge: https://github.com/conda-forge/unstructured-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/unstructured/0.15.8
- pypi inspector: https://inspector.pypi.io/project/unstructured/0.15.8

### Depends on

- https://github.com/AnacondaRecipes/unstructured-client-feedstock/pull/1

### Explanation of changes:

- version `0.15.8` (note: can't bump to the latest `0.15.9` because `python-oxmsg` is missing and was not estimated for SF)
- upstream tests
- unfortunately, there are too many places to fix for building for python 3.8.

[PKG-4794]: https://anaconda.atlassian.net/browse/PKG-4794?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Jrice1317 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,71 +6,119 @@ package:
...
 
 build:
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  # skip py312: langdetect not available
+  # skip s390x: deepdiff, httpx, unstructured-client not available
+  # skip win: python-magic not available
+  skip: true  # [py<39 or py>311 or s390x or win]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
   number: 0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:24]: Is there missing an entry_point `unstructured-ingest=unstructured.ingest.main:main`? See https://inspector.pypi.io/project/unstructured/0.15.8/packages/56/2d/a4cedd823b401e450053a6b8ed6088e3fd568d16bb1bc6d4f4ee5aca242e/unstructured-0.15.8.tar.gz/unstructured-0.15.8/setup.py#line.108

**Review by skupr-anaconda — COMMENTED** [review]
There are different pinnings and a list of required runtime dependencies:
- https://github.com/Unstructured-IO/unstructured/blob/0.15.8/setup.py#L110
- https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/base.in
- https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/deps/constraints.txt

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:24]: I've added it. It seems that it is optional for `unstructured-ingest`. I've added a run test, it requires `pandas`.

**Comment by lorepirri** [comment]
> There are different pinnings and a list of required runtime dependencies:
> 
> * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/setup.py#L110
> * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/base.in
> * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/deps/constraints.txt

thanks. `conda-forge` added all of those deps. I removed them (one option would be to move them in `run_constrained`) and used only the specified dependencies. Tests pass.

**Comment by lorepirri** [comment]
In general, there are a lot of optional dependencies here in this package. I don't know if it's worth adding all of them, considering that they all use the two files style (one for the dependencies, and one general one for the pinnings).

**Comment by skupr-anaconda** [comment]
> > There are different pinnings and a list of required runtime dependencies:
> > 
> > * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/setup.py#L110
> > * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/base.in
> > * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/deps/constraints.txt
> 
> thanks. `conda-forge` added all of those deps. I removed them (one option would be to move them in `run_constrained`) and used only the specified dependencies. Tests pass.

Moving them to `run_constrained` will be a better option

**Comment by lorepirri** [comment]
> > > There are different pinnings and a list of required runtime dependencies:
> > > 
> > > * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/setup.py#L110
> > > * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/base.in
> > > * https://github.com/Unstructured-IO/unstructured/blob/0.15.8/requirements/deps/constraints.txt
> > 
> > 
> > thanks. `conda-forge` added all of those deps. I removed them (one option would be to move them in `run_constrained`) and used only the specified dependencies. Tests pass.
> 
> Moving them to `run_constrained` will be a better option

done,
thanks for the help!

**Review by skupr-anaconda — APPROVED** [review]
Thanks!

**Pull request merged**